modules/tinymce/src/core/test/ts/webdriver/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,96 @@
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    add_unload_trigger: false,
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [ Theme ], true);
+  const platform = PlatformDetection.detect();
+
+  context('Page Up', () => {
+    it('TINY-4612: caret should be placed at the start of the line if the first child is an inline element', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 0);
+    });
+
+    it('TINY-4612: caret wont move if it is not at inline element at the start of the line', async function () {
+      // Page Up/Down on Mac scroll the viewport instead of change the caret position/selection
+      if (!platform.os.isOSX()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link2</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 4);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 4);
+    });
+
+    it('TINY-4612: caret wont move inline_boundaries: false', async function () {
+      if (!platform.os.isOSX()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.settings.inline_boundaries = false;
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 1 ], 2);
+      delete editor.settings.inline_boundaries;
+    });
+  });
+
+  context('Page Down', () => {
+    it('TINY-4612: caret should be placed at the end of the line if the last child is an inline element', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 2);
+    });
+
+    it('TINY-4612: caret wont move if it is not at inline element at the end of the line', async function () {
+      // Windows os default behavior (regardless the browser) is moving the caret at the end of the line
+      if (platform.os.isWindows()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 1 ], 2);
```

Comment:
This is also not an expected/correct assertion, as this should be at the end of the line. This is likely passing again due to it running on macOS which scrolls instead.

Reply:
Worth noting it'd probably be best to branch the assertions and assert the relevant state (ie macOS doesn't move, whereas other OS's do). At the moment throughout this test we're just completely skipping instead which isn't ideal.

Reply:
updated